### PR TITLE
[Choreolib-java] Updated auto chooser to not inject the factory

### PR DESCRIPTION
**Pull request by oh-yes-0-fps**
It was easier to just convert the internals to use commands and drop the autofactory dependency on the factory entirely.

### CODE REVIEW

**Comment by BR88C** [comment]
(Resolves #1011)

**Review by spacey-sooty — APPROVED** [review]

File: choreolib/src/main/java/choreo/auto/AutoChooser.java [hunk 1]
```diff
@@ -44,31 +44,31 @@ public class AutoChooser {
   private static final Alert selectedNonexistentAuto =
       Choreo.alert("Selected an auto that isn't an option", kError);
 
-  private final HashMap<String, Function<AutoFactory, AutoRoutine>> autoRoutines =
-      new HashMap<>(Map.of(NONE_NAME, factory -> AutoFactory.VOID_ROUTINE));
+  private final HashMap<String, Supplier<Command>> autoRoutines =
```
Comment (BR88C) [inline choreolib/src/main/java/choreo/auto/AutoChooser.java:47]: `AutoFactory.commandAsAutoRoutine()` can be removed, as it is no longer used

**Review by BR88C — CHANGES_REQUESTED** [review]

**Review by BR88C — APPROVED** [review]
LGTM

File: choreolib/src/main/java/choreo/auto/AutoChooser.java [hunk 2]
```diff
@@ -139,7 +140,7 @@ public void update() {
    * AutoChooser chooser;
    * Autos autos = new Autos(swerve, shooter, intake, feeder);
    * public Robot() {
-   *   chooser = new AutoChooser(Choreo.createAutoFactory(...), "/Choosers");
+   *   chooser = new AutoChooser("/Choosers");
    *   addPeriodic(chooser::update, 0.02); // chooser must be updated every loop
    *   // fourPieceRight is a method that accepts an AutoFactory and returns an AutoRoutine.
    *   chooser.addAutoRoutine("4 Piece right", autos::fourPieceRight);
```
Comment (Daniel1464) [inline choreolib/src/main/java/choreo/auto/AutoChooser.java:146]: Needs docs updates

File: choreolib/src/main/java/choreo/auto/AutoChooser.java [hunk 3]
```diff
@@ -168,60 +169,21 @@ public void addAutoRoutine(String name, Function<AutoFactory, AutoRoutine> gener
    * AutoChooser chooser;
    * Autos autos = new Autos(swerve, shooter, intake, feeder);
    * public Robot() {
-   *   chooser = new AutoChooser(Choreo.createAutoFactory(...), "/Choosers");
+   *   chooser = new AutoChooser("/Choosers");
    *   addPeriodic(chooser::update, 0.02); // chooser must be updated every loop
    *   // fourPieceLeft is a method that accepts an AutoFactory and returns a command.
    *   chooser.addAutoCmd("4 Piece left", autos::fourPieceLeft);
```
Comment (Daniel1464) [inline choreolib/src/main/java/choreo/auto/AutoChooser.java:175]: Ditto

File: choreolib/src/main/java/choreo/auto/AutoChooser.java [hunk 4]
```diff
@@ -238,18 +200,21 @@ public AutoRoutine getSelectedAutoRoutine() {
    *
    * @return A command that runs the selected {@link AutoRoutine}
    */
-  public Command autoSchedulingCmd() {
+  public Command selectedCommandScheduler() {
```
Comment (Daniel1464) [inline choreolib/src/main/java/choreo/auto/AutoChooser.java:203]: Ditto

**Review by Daniel1464 — CHANGES_REQUESTED** [review]

**Pull request merged**